pydantic/json_schema.py

```diff
@@ -1445,6 +1445,9 @@ def _update_class_schema(
                 f"model_config['json_schema_extra']={json_schema_extra} should be a dict, callable, or None"
             )
 
+        if hasattr(cls, '__deprecated__'):
+            json_schema['deprecated'] = True
+
```

Comment:
Just one thought here - we probably want to check if `cls.__deprecated__` is `True` or a `str`, right? You can imagine a case where someone sets `__deprecated__` to `False` on a model and this would lead to unexpected behavior. We maybe also want to check for `None`.